__tests__/endpoints.test.js

```diff
@@ -148,6 +148,35 @@ describe('NC news endpoint tests',()=>{
                     expect(body.msg).toBe('Invalid query')
                 })
             })
+            test('Respond with 200 with all the articles with a topic query',()=>{
+                return request(app)
+                .get('/api/articles?topic=mitch')
+                .expect(200)
+                .then(({body})=>{
+                    expect(body.articles.length).toBeGreaterThan(0)
+                        body.articles.forEach(article=>{
+                            expect(article.topic).toBe('mitch')
+                            expect(typeof article.article_id).toBe('number');
+                            expect(typeof article.title).toBe('string');
+                            expect(typeof article.author).toBe('string');
+                            expect(typeof article.topic).toBe('string');
+                            expect(typeof article.created_at).toBe('string');
+                            expect(typeof article.votes).toBe('number');
+                            expect(typeof article.article_img_url).toBe('string');
+                            expect(typeof article.comment_count).toBe('number');
+                        })
+                    expect(body.articles).toBeSortedBy('created_at', {descending: true})
+                })
+            })
+            test('Respond with 400 when topic queried is not greenlighted',()=>{
+                return request(app)
+                .get('/api/articles?sort_by=title&order=asc&topic=memes')
+                .expect(400)
+                .then(({body})=>{
+                    expect(body.msg).toBe('Bad Request')
+
+                })
```

Comment:
One untested behaviour here:
We have a topic with no associated articles (topic=paper)
In this case we would expect 200 responds with empty array of articles